Parsedown.php

```diff
@@ -541,7 +546,19 @@ protected function blockList($Line)
 
     protected function blockListContinue($Line, array $Block)
     {
-        if ($Block['indent'] === $Line['indent'] and preg_match('/^'.$Block['pattern'].'(?:[ ]+(.*)|$)/', $Line['text'], $matches))
+        if (
+            (
+                    $Block['indent'] === $Line['indent'] 
+                and $Block['data']['type'] === 'ol'
+                and preg_match('/^'.$Block['pattern'].'(?:[ ]+(.*)|$)/', $Line['text'], $matches)
+            )
+            or
+            (
+                    $Block['indent'] === $Line['indent']
+                and $Block['data']['type'] === 'ul'
+                and preg_match('/^'.preg_quote($Block['data']['matchText']).'(?:[ ]+(.*)|$)/', $Line['text'], $matches)
+            )
```

Comment:
This can be summarised as the following: 

* If either of the following renders true, then add the item as normal
  * If the list is the same indent as the current block
  * and the type is ordered
  * check to see if the line matches the expected pattern

  or
  * If the list is the same indent as the current block
  * and the type is un-ordered
  * check to see if the line has the same list marker (in accordance with CommonMark spec)
* otherwise return null if the following is true
 * the line has the same indent as the current block